storage/tsdb/tsdb.go

```diff
@@ -151,51 +151,14 @@ func Open(path string, l log.Logger, r prometheus.Registerer, opts *Options) (*t
 
 // StartTime implements the Storage interface.
 func (a adapter) StartTime() (int64, error) {
-	startTime := int64(model.Latest)
+	var startTime int64
 
-	var indexr tsdb.IndexReader
 	if len(a.db.Blocks()) > 0 {
-		var err error
-		indexr, err = a.db.Blocks()[0].Index()
-		if err != nil {
-			return startTime, err
-		}
+		startTime = a.db.Blocks()[0].Meta().MinTime
```

Comment:
This is going to be too far back by a block

Reply:
According to @fabxc in #3340 it won't because we add the safety margin (`--storage.tsdb.min-block-duration`). Then AFAIK the oldest block is the result of compaction of several smaller blocks. Therefore the difference between `blocks[0].MinTime` and the real first timestamp in the tsdb should be smaller than `min-block-duration`.
Is that right ?

Reply:
Yes. Where is the safety margin being added though, or rather, where is `a.startTimeMargin` being set and to what value?

Reply:
https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L329

Reply:
Thanks, that makes sense AFAICS. @brian-brazil any concerns still?

Reply:
I think we need to double the minimum then. If there's a gap of min-block-duration to the first data point then we still need another block duration to cover a Prometheus that ingests both fresh and maximally-old data.

Reply:
Mh, sounds right, 1.5x would be enough but 2x seems safer.
However, it's a heuristic at best anyway. We agree on that, right?

Reply:
It's a heuristic, but we should be choosing a conservative value to avoid weirdness around the boundary of this setting. It's better to request more data form remote storage than to miss some data due to ingestion with timestamps.

Reply:
Yes, totally agreeing. All good then.